src/Cli/dotnet/commands/dotnet-new/NewCommandParser.cs

```diff
@@ -105,7 +114,75 @@ private static ITemplateEngineHost CreateHost(bool disableSdkTemplates)
 
         private static bool RestoreProject(string pathToRestore)
         {
-            return RestoreCommand.Run(new string[] { pathToRestore }) == 0;
+            try
+            {
+                PathUtility.EnsureAllPathsExist(new[] { pathToRestore }, CommonLocalizableStrings.FileNotFound, allowDirectories: true);
+                return RestoreCommand.Run(new string[] { pathToRestore }) == 0;
+            }
+            catch (Exception e)
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.RestoreCallback_Failed, e.Message));
+                return false;
+            }
+        }
+
+        private static bool AddPackageReference(string projectName, string packageName, string version)
+        {
+            try
+            {
+                PathUtility.EnsureAllPathsExist(new[] { projectName }, CommonLocalizableStrings.FileNotFound, allowDirectories: false);
+                string command = $"add {projectName} package {packageName}";
```

Comment:
I concur - the `string[]` form of `Parse` should be used instead specifically because it handles escaping correctly. Same for all the new commands introduced in this PR.